src/init.hpp

```diff
@@ -7,7 +7,7 @@ namespace gpudrive
 {
 
     // Constants computed from train files.
-    constexpr size_t MAX_OBJECTS = 515;
+    constexpr size_t MAX_OBJECTS = consts::kMaxAgentCount;
     constexpr size_t MAX_ROADS = 956;
     constexpr size_t MAX_POSITIONS = 91;
     constexpr size_t MAX_GEOMETRY = 1746;
```

Comment:
This is interesting; why is here this set by consts and the others not set by consts? Additionally, I know that's not the point of this PR, but these could use some comments for what each is because the variable name is not sufficiently clear i.e. MAX_POSITIONS is actually the available length of expert data but you couldn't infer that from that name. So, maybe we should also open an issue that these should be renamed to something more explicit.

Reply:
Yes you are right. I calculated these constants by going through the dataset. These are the max objects and roads we will see if we were to read the entire dataset. It ensures we dont miss anything. 

We have max objects as a number before because we were not limited to initializing to 128 agents (kMaxAgents) before. But now that we do it, we can just set the const value as we are not reading more than 128 agents anyways. 

Max roads are just how many roads we can expect, max geometry is how many points can a road object have in its poly line.